### ISSUE TITLE
Lenovo Thinkpad P14s Speaker and built in mic does not work ubuntu 22.04

### ISSUE THREAD

**Issue by Ravis22** [thesofproject/linux#4073]
**Describe the bug**
I upgraded from ubuntu 20.04 to 22.04, everything was working fine. However after one day taking latest updates from software updater. After rebooting laptop speakers and mic stopped working. 
After doing google and looking at some posts I was able to make my speakers to work with below changes:
`options snd-intel-dspcfg dsp_driver=1`
With the above change my internal mic does not work only speakers work.
I also tried to download fresh ubuntu 22.04 and launched it in try mode. There also speakers and mic did not show. 

**To Reproduce**
Install latest ubuntu 22.04 and check speaker and mic not working

**Reproduction Rate**
100%

**Expected behavior**
Mic and speaker should work without any manual changes in the .conf files

**Impact**
since I work from home and have to attend meetings my work is impacted.

**Environment**
1) Branch name and commit hash of the 2 repositories: sof (firmware/topology) and linux (kernel driver).
    * Kernel: v 5.15.0-56-generic
    * SOF: firmware-sof-signed/jammy-updates,jammy-updates,now 2.0-1ubuntu4 all 
2) Name of the topology file
    * Topology: Not sure how to find it
3) Name of the platform(s) on which the bug is observed.
    * Platform: Ubuntu 22.04

See the screenshot 
![Screenshot from 2022-12-05 18-28-37](https://user-images.githubusercontent.com/35861100/205644667-ac15814c-85e1-43aa-b203-e9235574a9d7.png)

**dmesg logs** after executing: `sudo alsa force-reload` 	

```
[ 7481.282273] sof-audio-pci-intel-cnl 0000:00:1f.3: DSP detected with PCI class/subclass/prog-if info 0x040380
[ 7481.282292] sof-audio-pci-intel-cnl 0000:00:1f.3: Digital mics found on Skylake+ platform, using SOF driver
[ 7481.282438] sof-audio-pci-intel-cnl 0000:00:1f.3: DSP detected with PCI class/subclass/prog-if 0x040380
[ 7481.282554] sof-audio-pci-intel-cnl 0000:00:1f.3: bound 0000:00:02.0 (ops i915_audio_component_bind_ops [i915])
[ 7481.288327] sof-audio-pci-intel-cnl 0000:00:1f.3: use msi interrupt mode
[ 7481.300724] sof-audio-pci-intel-cnl 0000:00:1f.3: hda codecs found, mask 5
[ 7481.300727] sof-audio-pci-intel-cnl 0000:00:1f.3: using HDA machine driver skl_hda_dsp_generic now
[ 7481.300732] sof-audio-pci-intel-cnl 0000:00:1f.3: DMICs detected in NHLT tables: 2
[ 7481.300888] sof-audio-pci-intel-cnl 0000:00:1f.3: Firmware info: version 2:0:0-b678a
[ 7481.300889] sof-audio-pci-intel-cnl 0000:00:1f.3: Firmware: ABI 3:20:0 Kernel ABI 3:18:0
[ 7481.300891] sof-audio-pci-intel-cnl 0000:00:1f.3: warn: FW ABI is more recent than kernel
[ 7481.300893] sof-audio-pci-intel-cnl 0000:00:1f.3: unknown sof_ext_man header type 3 size 0x30
[ 7482.206249] sof-audio-pci-intel-cnl 0000:00:1f.3: cl_dsp_init: timeout with rom_status_reg (0x80000) read
[ 7482.206295] sof-audio-pci-intel-cnl 0000:00:1f.3: error: extended rom status:  0x6000021 0x0 0x0 0x0 0x0 0x0 0x1811102 0x0
[ 7482.206323] sof-audio-pci-intel-cnl 0000:00:1f.3: error: dsp init failed after 3 attempts with err: -110
[ 7482.206363] sof-audio-pci-intel-cnl 0000:00:1f.3: error: failed to reset DSP
[ 7482.206367] sof-audio-pci-intel-cnl 0000:00:1f.3: error: failed to boot DSP firmware -110
[ 7482.207764] sof-audio-pci-intel-cnl 0000:00:1f.3: error: sof_probe_work failed err: -110

```

**Comment by plbossart** [comment]
https://github.com/ryanprescott/realtek-verb-tools/wiki/How-to-sniff-verbs-from-a-Windows-sound-driver

**Comment by plbossart** [comment]
Your firmware is 
````
Firmware info: version 2:0:0-b678a
````

so 2.0. Try updating to 2.2.
https://launchpad.net/ubuntu/+source/firmware-sof

**Comment by Ravis22** [comment]
Upgraded to 2.2 from [here](https://launchpad.net/ubuntu/+source/firmware-sof/2.2.2-1ubuntu1/+build/24589645) 
Still getting no sound or mic. Same Dummy output.
Also as per the information available [here](https://launchpad.net/ubuntu/+source/firmware-sof) for JellyFish v2.0 is supported. 

Need help to resolve the issue. Attaching dmesg logs after upgrading to 2.2
[dmesg.txt](https://github.com/thesofproject/linux/files/10162436/dmesg.txt)

**Comment by Ravis22** [comment]
closed by mistake

**Comment by Ravis22** [comment]
Update External (wired 3.5mm) mic and speaker both works on headphone
when I have `options snd-intel-dspcfg dsp_driver=1` added in `/etc/modprobe.d/alsa-base.conf`

**Comment by plbossart** [comment]
Looks like a firmware signature issue to me @Ravis22 @lgirdwood @ujfalusi 

````
[    4.638229] sof-audio-pci-intel-cnl 0000:00:1f.3: cl_dsp_init: timeout with rom_status_reg (0x80000) read
[    4.638260] sof-audio-pci-intel-cnl 0000:00:1f.3: error: extended rom status:  0x6000021 0x0 0x0 0x0 0x0 0x0 0x1811102 0x0
[    4.638274] sof-audio-pci-intel-cnl 0000:00:1f.3: error: dsp init failed after 3 attempts with err: -110
[    4.638287] sof-audio-pci-intel-cnl 0000:00:1f.3: error: failed to reset DSP
[    4.638288] sof-audio-pci-intel-cnl 0000:00:1f.3: error: failed to boot DSP firmware -110
[    4.641422] sof-audio-pci-intel-cnl 0000:00:1f.3: error: sof_probe_work failed err: -110
````

**Comment by plbossart** [comment]
Adding @mrhpearson on the Lenovo side

**Comment by plbossart** [comment]
I would check if the BIOS was updated and if the CSE is indeed enabled https://thesofproject.github.io/latest/getting_started/intel_debug/suggestions.html#make-sure-the-me-is-enabled

**Comment by Ravis22** [comment]
BIOS was not updated. It stopped working after taking updates from software updater application. 
I am not sure how to check if CSE is enabled on disabled. Guide me how to check, I'll share the updates

**Comment by mrhpearson** [comment]
Hi - which P14s is this please? Will see if we can confirm here
Mark

**Comment by Ravis22** [comment]
See attached file if that helps. If not please let me know where can I find the information you are looking for.
 
![WhatsApp Image 2022-12-07 at 5 27 10 AM](https://user-images.githubusercontent.com/35861100/206051696-039ca77f-cee3-4606-9924-94a60adaf893.jpeg)

**Comment by mrhpearson** [comment]
OK - P14s G1 :) Thanks.

Your BIOS is old - version 1.26 is released on LVFS (and EC 1.14). Recommend upgrading; though I don't know if that will be related to the audio issue but worth doing.

I'll see if we can reproduce but it's a bit of a nightmare this week as we have team members on PTO and others in lockdown and unable to go to the office. I don't have this platform myself unfortunately

**Comment by Ravis22** [comment]
Everything was working fine till Nov 30. After I took software updates suggested by Ubuntu 22.04 audio/mic and bluetooth stopped working. 
Upgrading to BIOS might not be an option for me. 
We can look at this in next week.

**Comment by Ravis22** [comment]
I was able to fix the issue as explained [here](https://forums.gentoo.org/viewtopic-p-8752857.html?sid=a9ca4dbf89312124eaaf707fed943025)
My dmesg error log was similar to the errors reported in [this](https://github.com/thesofproject/sof/issues/2842) bug. 
I tried below command and the speakers and mic started working post reboot. I do not see any error in dmesg 
`fwupdmgr reinstall <Intel ME id>`

I got the device id using below command
`fwupdmgr get-history` which gave me the history of recent updates. I noticed there was an update for Intel ME


Thank you for your time and support.
